From 5bef29dd5b153e63f02604cc058e1270798d3d5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 22 May 2026 14:35:17 +0000
Subject: [PATCH] Fix detecting extensions

While `format_filter_get_extension()` returns a string with `.` as first
character (stored in `_combinedExtensions`), `filter_get_extension()`
does not have dots (stored in `_extensions`).

Also, add `.zst` to the `doubleExtensions` list.
---
 app/autoAr.js           | 6 +++---
 app/desktopIconsUtil.js | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/app/autoAr.js b/app/autoAr.js
index cac7143c..53219078 100644
--- a/app/autoAr.js
+++ b/app/autoAr.js
@@ -155,11 +155,11 @@ var AutoAr = class {
     }
 
     extensionIsAvailable(extension) {
-        return (extension in this._extensions) || (extension in this._combinedExtensions);
+        return (`.${extension}` in this._extensions) || (extension in this._combinedExtensions);
     }
 
     getFormatAndFilterForExtension(extension) {
-        if (extension in this._extensions) {
+        if (`.${extension}` in this._extensions) {
             return this._extensions[extension];
         }
         if (extension in this._combinedExtensions) {
@@ -170,7 +170,7 @@ var AutoAr = class {
 
     _getFormatAndFilterForFilename(fileName) {
         for (let extension in this._combinedExtensions) {
-            if (fileName.endsWith(`.${extension}`)) {
+            if (fileName.endsWith(extension)) {
                 return this._combinedExtensions[extension];
             }
         }
diff --git a/app/desktopIconsUtil.js b/app/desktopIconsUtil.js
index 3f084dd7..649f73ea 100644
--- a/app/desktopIconsUtil.js
+++ b/app/desktopIconsUtil.js
@@ -263,7 +263,7 @@ function getFileExtensionOffset(filename, opts = {'isDirectory': false}) {
     let offset = filename.length;
     let extension = '';
     if (!opts.isDirectory) {
-        const doubleExtensions = ['.gz', '.bz2', '.sit', '.Z', '.bz', '.xz'];
+        const doubleExtensions = ['.gz', '.bz2', '.sit', '.Z', '.bz', '.xz', '.zst'];
         for (const item of doubleExtensions) {
             if (filename.endsWith(item)) {
                 offset -= item.length;
-- 
GitLab

